pyrtl/simulation.py

```diff
@@ -793,15 +793,44 @@ def add_fast_step(self, fastsim):
         for wire_name in self.trace:
             self.trace[wire_name].append(fastsim.context[wire_name])
 
-    def print_trace(self, file=sys.stdout):
-        # TODO: make multi-digit values more readable and implement better alignment (issue #203)
+    def print_trace(self, file=sys.stdout, base=10, compact=False):
+        """
+        Prints a list of wires and their current values.
+        :param int base: the base the values are to be printed in
+        :param bool compact: whether to omit spaces in output lines
+        """
         if len(self.trace) == 0:
             raise PyrtlError('error, cannot print an empty trace')
-        maxlen = max(len(w) for w in self.trace)
-        for w in sorted(self.trace, key=_trace_sort_key):
-            file.write(' '.join([w.rjust(maxlen),
-                       ''.join(str(x) for x in self.trace[w])+'\n']))
-            file.flush()
+        if base not in (2, 8, 10, 16):
+            raise PyrtlError('please choose a valid base')
+
+        def dec(i):
+            return '0d' + str(i)
+
+        def octal(i):
+            return ('0o' + oct(i)[(0 if i == 0 else 1):] if sys.version_info.major == 2
+                    else oct(i))
+
+        bfunc = {2: bin, 8: octal, 10: dec, 16: hex}[base]
```

Comment:
I'm wondering if using `string.format` will make this code and the one below,
`line += bfunc(x)[2:].rjust(maxlenval+1) `
look simpler
https://docs.python.org/2/library/string.html#format-examples

The nice part is that with it, you don't have the problem of stripping prefixes like `0x`